frontend/main.js

```diff
@@ -271,9 +264,14 @@ async function startApp () {
           this.initOrResetPeriodicNotifications()
           this.checkAndEmitOneTimeNotifications()
         }
-
         // NOTE: should set IdleVue plugin here because state could be replaced while logging in
         Vue.use(IdleVue, { store, idleTime: 2 * 60 * 1000 }) // 2 mins of idle config
+
+        // NOTE: the SignupForm and LoginForm call requestNotificationPermission() upon LOGIN_COMPLETE,
+        // which also results in a call to 'service-worker/setup-push-subscription'.
```

Comment:
I've said in a different comment why `LOGIN_COMPLETE` can't be used for this, but to reiterate it here: the notification permission needs to be requested from direct user interaction. If it's not, or if it takes too long between the user interaction and the permission request, the browser will make it fail.